src/include/simeng/OS/MemRegion.hh

```diff
@@ -82,13 +83,21 @@ class MemRegion {
   bool overlapsStack(uint64_t addr, size_t size);
 
   /** This method retrieves the VMA containing addr. */
-  VirtualMemoryArea* getVMAFromAddr(uint64_t addr);
+  VirtualMemoryArea getVMAFromAddr(uint64_t addr);
+
+  /** This method returns the shared_ptr to the VMAlist. */
+  std::shared_ptr<std::list<VirtualMemoryArea>> getVmaList();
 
   /** This method retrieves the VMA head. */
-  VirtualMemoryArea* getVMAHead() { return vm_head_; };
+  VirtualMemoryArea getVMAHead() {
```

Comment:
This is only used for the test suite but I don't think only two functions i.e  (getVmaList and getVmaSize) justify the creation of a friend class. Given that this will return a copy of the head, this function is not harmful and will not change the actually head of the list.